src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -354,17 +355,36 @@ class LedgerMasterImp
             mCurrentLedger.getMutable();
         int recovers = 0;
 
-        for (auto const& it : mHeldTransactions)
+    #if RIPPLE_OPEN_LEDGER
+        getApp().openLedger().modify(
+            [&](OpenView& view, beast::Journal j)
+            {
+                bool any = false;
+                for (auto const& it : mHeldTransactions)
+                {
+                    ApplyFlags flags = tapNONE;
+                    if (getApp().getHashRouter().addSuppressionFlags (it.first.getTXID (), SF_SIGGOOD))
```

Comment:
`addSuppressionFlags` - this needs some more commenting here, I'm really unsure what's going on.